src/ZkFold/Base/Algebra/Basic/Field.hs

```diff
@@ -205,10 +228,12 @@ instance (Field f, Eq f, IrreduciblePoly f e) => MultiplicativeGroup (Ext3 f e)
     invert (Ext3 a b c) =
         let (g, s) = eea (toPoly [a, b, c]) (irreduciblePoly @f @e)
         in case scaleP (one / lt g) 0 s of
-            P (x:y:z:_) -> Ext3 x y z
-            P [x, y]    -> Ext3 x y zero
-            P [x]       -> Ext3 x zero zero
-            P []        -> Ext3 zero zero zero
+            P []     -> Ext3 zero zero zero
+            P [x]    -> Ext3 x zero zero
+            P [x, y] -> Ext3 x y zero
+            P v      -> Ext3 (v V.! 0) (v V.! 1) (v V.! 2)
+
+instance (Field f, Eq f, IrreduciblePoly f e) => Field (Ext3 f e) where
```

Comment:
`Ext3 f e` has at least the same roots of unity as `f`

Reply:
Yeah, I thought about the same. So for Ext2 and Ext3, will `rootOfUnity n = (\r -> Ext2 r zero) <$> rootOfUnity n` still be a primitive root of unity?

Reply:
Yes! I think so